File: app/src/main/java/fup/prototype/robprototype/view/ViewModelFactory.java
package fup.prototype.robprototype.view;

import android.support.annotation.NonNull;

import fup.prototype.robprototype.model.Repository;
import fup.prototype.robprototype.view.adapters.RepositoryItemViewModel;

public final class ViewModelFactory  {

    public static RepositoryItemViewModel create(@NonNull final Repository repository){
        final RepositoryItemViewModel viewModel = new RepositoryItemViewModel();
        viewModel.repoName.set(repository.getName());
        return viewModel;
    }
}

File: app/src/main/java/fup/prototype/robprototype/view/ViewModelFactory.kt
package fup.prototype.robprototype.view

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.rxdroid.repository.repositories.detail.DetailsUiRepository
import com.rxdroid.repository.repositories.search.UserUiRepository
import fup.prototype.robprototype.details.DetailViewModel
import fup.prototype.robprototype.search.SearchViewModel
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class ViewModelFactory @Inject constructor(private val userUiRepo: UserUiRepository,
                                           private val detailsUiRepo: DetailsUiRepository)
    : ViewModelProvider.Factory {


    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(SearchViewModel::class.java)) {
            return SearchViewModel(userUiRepo) as (T)
        }
        if (modelClass.isAssignableFrom(DetailViewModel::class.java)) {
            return DetailViewModel(detailsUiRepo) as (T)
        }
        throw IllegalArgumentException("Unknown ViewModel class")

    }
}